src/main/java/de/dennisguse/opentracks/sensors/UintUtils.java

```diff
@@ -14,17 +14,20 @@ private UintUtils() {
      * @return diff
      */
     public static long diff(long a, long b, final long UINT_MAX) {
-        if (a < 0 || b < 0) {
-            throw new RuntimeException("a or b cannot be less than zero.");
+        
+        a %= (UINT_MAX + 1);
```

Comment:
Crashing on overflow is intended.
This will show if something else in the whole Bluetooth parsing is broken: crash early.
Otherwise a problem might be hidden for ages.